diff --git a/node_modules/@strapi/plugin-users-permissions/server/controllers/auth.js b/node_modules/@strapi/plugin-users-permissions/server/controllers/auth.js
index 02821f7..386d1d0 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/controllers/auth.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/controllers/auth.js
@@ -222,7 +222,18 @@ module.exports = {
     // Generate random token.
     const userInfo = await sanitizeUser(user, ctx);
 
-    const resetPasswordToken = crypto.randomBytes(64).toString('hex');
+    // const resetPasswordToken = crypto.randomBytes(64).toString('hex');
+
+    // Generate a 6-digit numeric OTP
+    const otpLength = 4;
+    const min = Math.pow(10, otpLength - 1);
+    const max = Math.pow(10, otpLength) - 1;
+    const resetPasswordTokenNumber = Math.floor(Math.random() * (max - min + 1)) + min;
+      
+    // Convert the numeric OTP to a string
+    const resetPasswordToken = resetPasswordTokenNumber.toString();
+      
+    // Use the generated string OTP in your code as needed
 
     const resetPasswordSettings = _.get(emailSettings, 'reset_password.options', {});
     const emailBody = await getService('users-permissions').template(
